src/Nacos.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Ocelot.Values;
+using Service = Ocelot.Values.Service;
+
+namespace Ocelot.Discovery.Nacos;
+
+public class Nacos : IServiceDiscoveryProvider
+{
+    private readonly INacosNamingService _client;
+    private readonly string _serviceName;
+
+    public Nacos(string serviceName,INacosNamingService client)
+    {
+        _client = client;
+        _serviceName = serviceName;
+    }
+
+    public async Task<List<Service>> GetAsync()
+    {
+        try
+        {
+            var instances = await _client.GetAllInstances(_serviceName)
+                .ConfigureAwait(false);
+
+            return instances?
+                .Where(i => i.Healthy && i.Enabled && i.Weight > 0) // 健康检查过滤
+                .Select(TransformInstance)
+                .ToList() ?? new List<Service>();
```

Comment:
```suggestion
                .ToList() ?? new();
```
Please remove the Chinese comment on the `Where` line. If you wish you may leave English comments